contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,81 @@
+# Priority in ResourceQuota
+
+@resouer
+
+Aug 2017
+  * [Objective](#objective)
+    * [Non-Goals](#non-goals)
+  * [Background](#background)
+  * [Overview](#overview)
+  * [Detailed Design](#detailed-design)
+    * [Expected behavior of ResourceQuota apply to pod](#resourcequota-apply-to-pod)
+    * [Expected behavior of ResourceQuota admission controller](#resourcequota-admission-controller)
+
+
+## Objective
+
+This feature is designed to make ResourceQuota become priority aware, three sub-tasks are included.
+
+* Add a field to ResourceQuota that specifies a priority class name.
```

Comment:
Ideally we would just have the PriorityClassName. I can imagine undesirable things happening if the cluster admin changes the mapping from PriorityClassName to priority non-atomically with changing the quotas that use that PriorityClassName (and since we don't have transactions, it would have to be non-atomic). Of course it would be brief, until they are in sync again.

The one reason I can think of to also include priority here is so that we can handle pods that have an integer priority and no PriorityClassName. But maybe to handle that case, we could say we allow ResourceQuota with empty PriorityClassName. The rule would be that such a ResourceQuota matches pods that do not have a PriorityClassName. (Pods' integer priorities would be ignored.) I think this might be the easiest thing for users to understand -- the rule would be "PriorityClassName in ResourceQuota is matched against PriorityClassName in the Pod, include the case where PriorityClassName is not set." And there would be no need for an integer priority in ResourceQuota.

Does that sound reasonable?